app/src/main/java/com/portfolio/tasky/entry/viewModels/EntryViewModel.kt

```diff
@@ -0,0 +1,12 @@
+package com.portfolio.tasky.entry.viewModels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import com.portfolio.tasky.usecases.NetworkStatus
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class EntryViewModel @Inject constructor(networkStatus: LiveData<NetworkStatus>) : ViewModel() {
+    val networkStatusObserver : LiveData<NetworkStatus> = networkStatus
+}
```

Comment:
Here's my recommendation how I would handle observing network status while following clean architecture:
Listening to these status changes is part of the data layer, since you interact with an API (Android's NetworkManager API which is a system service). Therefore the NetworkObserver should go in your data package. However, since in clean architecture the presentation layer (where your ViewModel is in) isn't allowed to talk to data, what you need is an abstraction in domain. Here a sample how that could look like:
```
// In domain
interface NetworkObserver {
    val networkStatus: LiveData<NetworkStatus>
}

// In data
class NetworkObserverImpl(private val context: Context): NetworkObserver {
    override val networkObserver: LiveData<NetworkStatus> = NetworkChangedReceiver(context)
}

// In your VM
class EntryViewModel @Inject constructor(private val observer: NetworkObserver): ViewModel() { ... }

// In your AppModule.kt
@Provides
@Singleton
fun provideNetworkObserver(@ApplicationContext context: Context): NetworkObserver {
    return NetworkObserverImpl(context)
}
```

Why do we do all this abstraction stuff? When we test stuff in Android, there are 2 types of tests we normally have:
1. Local unit tests which run directly on the JVM and don't need an Android device. These are very fast, but only work if they don't use Android specific dependencies such as Context (which your NetworkChangedReceiver needs)
2. Instrumented tests which can use such Android dependencies, but need to run on an emulator which makes them very slow.

You want to test as much as possible with local unit tests, especially ViewModel logic. That's why we need to keep our ViewModels free from Android dependencies or classes which need Android dependencies. By creating an interface for our NetworkObserver which we pass to the ViewModel, we could create a fake version of it for testing, since it doesn't force us to pass the NetworkObserverImpl. We could instead pass a version which doesn't need the context and just simulates the network status changes. Quite complex topic, in the video portal you'll find a whole playlist on testing. If it's unclear, please ask in WhatsApp 👍